packages/rich-text/src/plugins/List/index.tsx

```diff
@@ -149,6 +152,60 @@ export const withListOptions: CustomSlatePluginOptions = {
   },
 };
 
+const emptyNodeOfType = (type) => ({ type, children: [], data: {} });
+
+const isList = (node: CustomElement) =>
+  [BLOCKS.OL_LIST, BLOCKS.UL_LIST].includes(node.type as BLOCKS);
+
+const isListItem = (node: CustomElement) => node.type === BLOCKS.LIST_ITEM;
+
+const hasListAsDirectParent = (editor: Editor, path: Path) => {
+  const [parentNode] = (getParent(editor, path) || []) as NodeEntry;
+  return isList(parentNode as CustomElement);
+};
+
+const isValidInsideList = (node: CustomElement) =>
+  (LIST_ITEM_BLOCKS as Array<TopLevelBlockEnum | INLINES>)
+    .concat(Object.values(INLINES))
+    .includes(node.type as TopLevelBlockEnum);
+
+const unwrapInvalidGrandchildren = (editor: Editor, path: Path) => {
+  for (const [grandchild, grandchildPath] of Node.children(editor, path)) {
+    if (!isValidInsideList(grandchild as CustomElement)) {
+      Transforms.unwrapNodes(editor, { at: grandchildPath });
+    }
+  }
+};
+
+/**
+ * Ensures each list item follows the list schema.
+ */
+const normalizeList = (editor: Editor, path: Path) => {
+  for (const [child, childPath] of Node.children(editor, path)) {
+    if (Element.isElement(child)) {
+      unwrapInvalidGrandchildren(editor, childPath);
+      if (!isListItem(child)) {
+        Transforms.wrapNodes(editor, emptyNodeOfType(BLOCKS.LIST_ITEM), { at: childPath });
```

Comment:
I believe we should exit the loop early here, otherwise, this code will likely crash when more than one list item is invalid. The same applies to other normalize* functions

Read about [Multi-pass normalization](https://docs.slatejs.org/concepts/11-normalizing#multi-pass-normalizing) here

Reply:
You may find the `withNormalizer` helper from [this PR](https://github.com/contentful/field-editors/pull/944) useful